src/lib/Screen.re

```diff
@@ -92,6 +92,59 @@ let resize = (~rows, ~columns, model) => {
   };
 };
 
+let getColor =
+    (~theme, ~defaultBackground, ~defaultForeground, color: Vterm.Color.t) => {
+  let bg =
+    switch (defaultBackground) {
+    | Some(v) => v
+    | None => theme(0)
+    };
+
+  let fg =
+    switch (defaultForeground) {
+    | Some(v) => v
+    | None => theme(15)
+    };
+
+  switch (color) {
+  | DefaultBackground => bg
+  | DefaultForeground => fg
+  | Rgb(r, g, b) =>
+    if (r == 0 && g == 0 && b == 0) {
+      bg;
+    } else if (r == 240 && g == 240 && b == 240) {
+      fg;
+    } else {
+      Revery.Color.rgb_int(r, g, b);
+    }
+  | Index(idx) => theme(idx)
+  };
+};
+
+let getForegroundColor =
+    (
+      ~defaultBackground=?,
+      ~defaultForeground=?,
+      ~theme,
+      cell: Vterm.ScreenCell.t,
+    ) => {
+  cell.reverse == 0
+    ? getColor(~theme, ~defaultBackground, ~defaultForeground, cell.fg)
+    : getColor(~theme, ~defaultBackground, ~defaultForeground, cell.bg);
```

Comment:
The signal to noise ratio here is pretty low. Separating out the significant part of the conditional would help quite a bit:

```suggestion
  let termColor = cell.reverse == 0 ? cell.fg : cell.bg;
  getColor(~theme, ~defaultBackground, ~defaultForeground, termColor);
```

It might also be possible to reduce the "noise" even more with partial application, although perhaps at a slight cost to clarity.

Reply:
Makes sense. Looks clearer to pull that conditional out. I moved `getColor` to an `Internal` module and used partial application too.